File: admin-ui/src/services/swagger/userApi.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request';

/** 创建用户 POST /api/users */
export async function createUser(body: API.CreateUserRequestDTO, options?: { [key: string]: any }) {
  return request<number>('/api/users', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 获取用户详情 GET /api/users/${param0} */
export async function getUserDetail(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getUserDetailParams,
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<API.UserDTO>(`/api/users/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 更新用户 PUT /api/users/${param0} */
export async function updateUser(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateUserParams,
  body: API.UpdateUserRequestDTO,
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<any>(`/api/users/${param0}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 修改当前用户密码 PUT /api/users/change-password */
export async function changeCurrentUserPassword(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.changeCurrentUserPasswordParams,
  options?: { [key: string]: any },
) {
  return request<any>('/api/users/change-password', {
    method: 'PUT',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 获取当前用户详情 GET /api/users/current */
export async function getCurrentUserDetail(options?: { [key: string]: any }) {
  return request<API.CurrentUserInfoDTO>('/api/users/current', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 获取当前用户权限列表 GET /api/users/current/permission-codes */
export async function getCurrentUserPermissionCodes(options?: { [key: string]: any }) {
  return request<string[]>('/api/users/current/permission-codes', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 禁用用户 PUT /api/users/disable/${param0} */
export async function disableUser(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.disableUserParams,
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<any>(`/api/users/disable/${param0}`, {
    method: 'PUT',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 启用用户 PUT /api/users/enable/${param0} */
export async function enableUser(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.enableUserParams,
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<any>(`/api/users/enable/${param0}`, {
    method: 'PUT',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 为用户授予角色 PUT /api/users/grant-roles/${param0} */
export async function grantRoles(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.grantRolesParams,
  body: number[],
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<any>(`/api/users/grant-roles/${param0}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 查询用户列表 POST /api/users/list */
export async function listUsers(body: API.QueryUserRequestDTO, options?: { [key: string]: any }) {
  return request<API.PageDTOUserDTO>('/api/users/list', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 重置用户密码 PUT /api/users/reset-password/${param0} */
export async function resetPassword(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.resetPasswordParams,
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<any>(`/api/users/reset-password/${param0}`, {
    method: 'PUT',
    params: { ...queryParams },
    ...(options || {}),
  });
}

File: admin-backend/application-api/src/main/java/com/xfrog/platform/application/permission/api/UserApi.java
package com.xfrog.platform.application.permission.api;

import com.xfrog.framework.dto.PageDTO;
import com.xfrog.platform.application.permission.api.dto.CreateUserRequestDTO;
import com.xfrog.platform.application.permission.api.dto.CurrentUserInfoDTO;
import com.xfrog.platform.application.permission.api.dto.QueryUserRequestDTO;
import com.xfrog.platform.application.permission.api.dto.UpdateUserRequestDTO;
import com.xfrog.platform.application.permission.api.dto.UserDTO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;


@Tag(name = "UserApi", description = "用户管理接口")
@RequestMapping("/api/users")
public interface UserApi {
    @PostMapping("/list")
    @Operation(summary = "查询用户列表")
    PageDTO<UserDTO> listUsers(@Valid @RequestBody QueryUserRequestDTO queryUserRequestDTO);

    @PostMapping
    @Operation(summary = "创建用户")
    Long createUser(@Valid @RequestBody CreateUserRequestDTO user);

    @PutMapping("/{userId}")
    @Operation(summary = "更新用户")
    void updateUser(@PathVariable(name = "userId") Long userId, @Valid @RequestBody UpdateUserRequestDTO user);

    @GetMapping("/{userId}")
    @Operation(summary = "获取用户详情")
    UserDTO getUserDetail(@PathVariable(name = "userId") Long userId);

    @PutMapping("/reset-password/{userId}")
    @Operation(summary = "重置用户密码")
    void resetPassword(@PathVariable(name = "userId") Long userId);

    @PutMapping("/disable/{userId}")
    @Operation(summary = "禁用用户")
    void disableUser(@PathVariable(name = "userId") Long userId);

    @PutMapping("/enable/{userId}")
    @Operation(summary = "启用用户")
    void enableUser(@PathVariable(name = "userId") Long userId);

    @PutMapping("/change-password")
    @Operation(summary = "修改当前用户密码")
    void changeCurrentUserPassword(@RequestParam String newPassword);

    @PutMapping("/grant-roles/{userId}")
    @Operation(summary = "为用户授予角色")
    void grantRoles(@PathVariable(name = "userId") Long userId, @RequestBody List<Long> roleIds);

    @GetMapping("/current")
    @Operation(summary = "获取当前用户详情")
    CurrentUserInfoDTO getCurrentUserDetail();

    @GetMapping("/current/permission-codes")
    @Operation(summary = "获取当前用户权限列表")
    List<String> getCurrentUserPermissionCodes();
}
